### DISCUSSION TITLE
Status of GUI support for dual-band setups? (2.4 & 5ghz)

### DISCUSSION THREAD

**Discussion by smanley** [RaspAP/raspap-webgui#1116] [General]
Hi;

Been testing RaspAP for awhile and am very happy. 

I have a large deployment coming up, and dual band support is a requirement. The existing setup is a bit of a kludge though, and non-linux people need to admin/check the system.

Is there any update on when this functionality will be added to the GUI config?

Thanks,
Steve

**Comment by billz** [comment]
Good suggestion. This is a popular request, so I'll carve out some time and prototype a UI.

**Comment by billz** [comment]
A dual band toggle enables an additional tab to configure a second interface.

<img width="328" alt="dual" src="https://user-images.githubusercontent.com/229399/152020649-3d53eaae-9794-420f-b9cd-c2049035976c.png">

The less trivial bit is using an instanced systemd service to handle multiple hostapd configs. To simplify things for the user, I'd consider a 'configure related services' option, or similar, to apply defaults for dhcp + dnsmasq. Lots of config/state switching involved.

**Comment by smanley** [comment]
Thanks. I'm happy to support the project and this is the only real major deficiency I've been dealing with.

Once the second interface is configured, though, is further editing required?

Right now, you basically clone the interface, set the parameters, then you have to restart manually e.g.:

 sudo hostapd /etc/hostapd/wlan0.conf /etc/hostapd/wlan1.conf

The problem with this is it screws up the start/stop interface logic in the gui.  My use case is simplified as I'm all bridged, though.

**Reply by billz** [reply]
> Once the second interface is configured, though, is further editing required?

Both [dnsmasq and dhcp](https://docs.raspap.com/multiple/#configure-dnsmasq) require changes to support this. If a user enables this mode without configuring these services, the second AP will fail to start. This is the thinking behind the 'configure related services' option that would apply a set of defaults, no different from how default settings are managed elsewhere in the app. These settings could of course be modified by the user via the DHCP server UI.

> The problem with this is it screws up the start/stop interface logic in the gui

You're right, the docs demonstrate running hostapd interactively with debug output. It is, after all, experimental 😉 
As mentioned above, there's a fair bit of refactoring required to instead run the instanced version of the hostapd service, `hostapd@.service` (distinguished by the '@'), which is configured differently. 

See https://github.com/RaspAP/raspap-insiders/pull/72
